### PR TITLE
Additional Spinner Variants and Normal Variant Deprecation

### PR DESCRIPTION
**Pull request by wdo3650**
#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [ ] Include a change request file if publishing 
- [x] New feature, bugfix, or enhancement
- [ ] Includes tests
- [ ] Documentation update

#### Description of changes

Adds extra small, small and medium variants to the Spinner as well as deprecates the normal variant.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Spinner/Spinner.Props.ts [hunk 1]
```diff
@@ -20,6 +20,30 @@ export interface ISpinnerProps extends React.Props<Spinner> {
...
+  /**
+   * 20px Spinner diameter
+   */
+  medium = 2,
+
+  /**
+   * @deprecated
+   * Deprecated and will be removed at >= 2.0.0. Use medium instead.
+   */
+  normal = 2,
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Spinner/Spinner.Props.ts]: Might need @dzearing to weigh in here, but changing normal from 0 to 2 and large from 1 to 3 could cause some problems. I.E. the numbers should never change (unless deprecated), and the numbers don't have to be in size order.

normal 0
large 1
xsmall 2
small 3
medium 4

This maintains backwards compat.

File: packages/office-ui-fabric-react/src/components/Spinner/Spinner.tsx [hunk 2]
```diff
@@ -14,7 +14,9 @@ export class Spinner extends React.Component<ISpinnerProps, any> {
     return (
       <div className={ css('ms-Spinner', className) }>
         <div className={ css('ms-Spinner-circle',
-          { 'ms-Spinner--normal': type === SpinnerType.normal },
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Spinner/Spinner.tsx]: to continue to support SpinnerType.normal, we'd need to leave this one in. I see the class is still there, so SpinnerType.normal should still add ms-Spinner--normal until fully removed.

Comment (wdo3650) [inline packages/office-ui-fabric-react/src/components/Spinner/Spinner.tsx]: Thanks for catching this.  This should be addressed.

File: packages/office-ui-fabric-react/src/components/Spinner/Spinner.tsx [hunk 3]
```diff
@@ -1,27 +1,35 @@
...
-          { 'ms-Spinner--large': type === SpinnerType.large })
+          { 'ms-Spinner--xSmall': size === SpinnerSize.xSmall },
+          { 'ms-Spinner--small': size === SpinnerSize.small },
+          { 'ms-Spinner--medium': size === SpinnerSize.medium },
+          { 'ms-Spinner--large': size === SpinnerSize.large },
+          { 'ms-Spinner--normal': type === SpinnerType.normal }, // TODO remove deprecated value at >= 2.0.0
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Spinner/Spinner.tsx]: Hmm it is unclear what "normal" would map to. Shouldn't we be mapping normal to one of the non-deprecated values?

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Spinner/Spinner.tsx]: eh it's fine.

**Pull request merged**